torchdynamo: add linear pointwise(binary) fusion kernel

XiaobingSuper opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86583
* #86582
* #86581

Support binary fusion of Linear with:

- add
- sub
- mul
- div

XiaobingSuper committed:
torchdynamo: add linear binary fusion kernel

[ghstack-poisoned]

XiaobingSuper committed:
Update on " torchdynamo: add linear binary fusion kernel"




[ghstack-poisoned]

XiaobingSuper committed:
Update on " torchdynamo: add linear pointwise(binary) fusion kernel"


Support binary fusion of Linear with:

- add
- sub
- mul
- div



[ghstack-poisoned]

XiaobingSuper committed:
Update on " torchdynamo: add linear pointwise(binary) fusion kernel"


Support binary fusion of Linear with:

- add
- sub
- mul
- div



[ghstack-poisoned]

jgong5 commented on aten/src/ATen/native/mkldnn/RegisterMkldnnOpContextClass.cpp:
```diff
@@ -37,6 +37,8 @@ TORCH_LIBRARY(mkldnn, m) {
 
   m.def(TORCH_SELECTIVE_SCHEMA(
       "mkldnn::_linear_pointwise(Tensor X, Tensor W, Tensor? B, str attr, Scalar?[] scalars, str? algorithm) -> Tensor Y"));
+  m.def(TORCH_SELECTIVE_SCHEMA(
+      "mkldnn::_linear_pointwise.binary(Tensor X, Tensor other, Tensor W, Tensor? B, str attr) -> Tensor Y"));
```
Shall we add an inplace version where `other` and `Y` are alias?

XiaobingSuper committed:
Update on " torchdynamo: add linear pointwise(binary) fusion kernel"


Support binary fusion of Linear with:

- add
- sub
- mul
- div



[ghstack-poisoned]

EikanWang commented:
@jansel , could you help review this PR for TorchInductor?

XiaobingSuper committed:
Update on " torchdynamo: add linear pointwise(binary) fusion kernel"


Support binary fusion of Linear with:

- add
- sub
- mul
- div



[ghstack-poisoned]

XiaobingSuper committed:
Update on " torchdynamo: add linear pointwise(binary) fusion kernel"


Support binary fusion of Linear with:

- add
- sub
- mul
- div



[ghstack-poisoned]

EikanWang commented:
@pytorchbot merge
